spi/src/main/java/com/flipkart/varadhi/spi/services/PartitionMessages.java

```diff
@@ -0,0 +1,17 @@
+package com.flipkart.varadhi.spi.services;
+
+import com.flipkart.varadhi.entities.Offset;
+import lombok.AllArgsConstructor;
+
+/**
+ * Represent a batch of messages polled from a particular partition.
+ */
+@AllArgsConstructor
+public class PartitionMessages<O extends Offset> {
```

Comment:
1. will below placed in spi or entities ?
2. should paritition be abstracted as well or how it will be converted into tech stack specific identifier e.g. in pulsar partitioned topic has its own identifier  ?

Reply:
`will below placed in spi or entities ?` - will evaluate.

Reply:
Can we keep this in spi itself? I am thinking that this class exists due to spi interfaces. If the interfaces are not there, then there is no need for this entity itself. This entity pertains to spi interfaces.

And about the partition abstraction, I think this particular works for kafka & pulsar at least. In case of non-partitioned scenario, the partition id can be -1. and the topic can refer to the relevant reference.

I am also thinking of platforms that dont even have partitioned concept. To support them in future, I think instead of coming up with 1 general interface that somehow fits all usecases, we can broadly define 2 feature capabalities. Partitioned vs non-partitioned.
partitioned stack like (kafka, pulsar google pub-sub lite) can fit in the model present in this PR.
whereas a non-partitioned stack (whcih also will require a different sharding model) can use a different set of interface.

This feature capability value will obviously come from the messagingStackProvider.

Reply:
Or we dont rely on partitions at all. that will work in all cases.